restructuredtext_lint/lint.py

```diff
@@ -80,9 +75,33 @@ def error_collector(data):
     return errors
 
 
-def lint_file(filepath, encoding=None):
+def lint(content, filepath=None, custom_directives=None, custom_roles=None):
+    """Lint reStructuredText and return errors
+
+    :param string content: reStructuredText to be linted
+    :param string filepath: Optional path to file, this will be returned as the source
+    :param list custom_directives: custom directives to ignore
+    :param list custom_roles: custom roles to ignore
+    :rtype list: List of errors. Each error will contain a line, source (filepath),
+        message (error message), and full message (error message + source lines)
+    """
+    if custom_directives is None:
+        custom_directives = []
+    if custom_roles is None:
+        custom_roles = []
+    if custom_directives or custom_roles:
+        with ignores.register_unregister_ignores(custom_directives,
+                                                 custom_roles):
+            return _lint(content, filepath=filepath)
+    else:
+        return _lint(content, filepath=filepath)
```

Comment:
To reduce complexity, can we relocate the fallbacks or `if` checks to be within the `register_unregister` call and always use `with`?

```python
def lint(content, filepath=None, custom_directives=None, custom_roles=None):
    with ignores.register_unregister_ignores(custom_directives, custom_roles):
        return _lint(content, filepath=filepath)
```